### ISSUE TITLE
Remove xapian support

### ISSUE THREAD

**Issue by jmontoyaa** [chamilo/chamilo-lms#2330]

**Comment by ywarnier** [comment]
This task should be cancelled and support for Xapian reinstated. There were several requests for a full-text indexation engine lately. e-learning is maturing and finding content within thousands of courses is starting to be an important topic.
As such, the title should be "Update Xapian support" :-)

**Comment by jmontoyaa** [comment]
It should be something "re write", I'm not sure about the state of xapian but probably it changed? 
Also there should be a better way to work with xapian and entities ...

**Comment by opensourcecompany** [comment]
xapian is still alive ;-)

**Comment by ywarnier** [comment]
Yeah, so Xapian is still pretty much alive and still the best option since *all* other options (as far as I know) to index full-text are using Java, and we don't want any Java on our Chamilo servers (because it's another technology with its own issues, that we cannot support at the same time as we do PHP). Xapian is in C/C++ and can run without having to install anything additional on the server than what is necessary for a normal Linux + PHP server.
The issue at the moment is not really compiling Xapian. The issue is that Xapian changed and the PHP to Xapian interface is not working the same way anymore and we have to rewrite the Xapian module in Chamilo and nobody wants to invest into anyone's time to do so, so we're stuck because we've got so many other things to do for Chamilo before that.

**Comment by ywarnier** [comment]
I'm closing this issue as there is demand for the feature in future versions (planned for 2.1). The code will remain dormant in 2.0
